doc/examples/rendering/traceSetsArnold.gfr

```diff
@@ -0,0 +1,348 @@
+__children["Backdrop"]["__uiBound"].setValue( imath.Box2f( imath.V2f( -10, 14.9502449 ), imath.V2f( 30, 46.6711006 ) ) )
+__children["Backdrop"]["__uiPosition"].setValue( imath.V2f( -1.96337986, -19.7618332 ) )
+__children["Backdrop1"]["title"].setValue( 'Adding objects to a trace set' )
+__children["Backdrop1"]["description"].setValue( 'Gaffer manages trace sets using the same nodes as normal sets. \n\nThe only difference is that the `render:` prefix is used to denote that it should be converted to a render trace set.\n\nThe SetSpheres node adds the two spheres to their own trace set using the set name `render:spheres`.' )
+__children["Backdrop1"]["__uiBound"].setValue( imath.Box2f( imath.V2f( -10, -7.14684677 ), imath.V2f( 30, 13.4110298 ) ) )
+__children["Backdrop1"]["__uiPosition"].setValue( imath.V2f( -34.6775818, -40.1869965 ) )
+__children["Backdrop2"]["title"].setValue( 'Ray tagging' )
+__children["Backdrop2"]["description"].setValue( 'As the spheres are in a custom trace set, we can create a reflective surface that uses this set to control what is visible.\n\nThe aiTraceSet Node inserted after the shading network, before it is assigned modifies the rays traced by the attached network such that they either include or exclude objects in the named trace sets.\n\nTry toggling the Inclusive checkbox to switch the mirror between showing only the spheres or exluding them.\n\n' )
+__children["Backdrop2"]["__uiBound"].setValue( imath.Box2f( imath.V2f( -10, -17.6545105 ), imath.V2f( 30, 9.90445709 ) ) )
+__children["Backdrop2"]["__uiPosition"].setValue( imath.V2f( -66.0190277, -68.9063416 ) )
+__children["Backdrop3"]["title"].setValue( 'Shadows!' )
+__children["Backdrop3"]["description"].setValue( "When inclusive is turned off on the aiTraceSet node, the spheres wil be excluded from the reflections, but their shadows still appear on the ground.\n\nThis is because the plane is shaded normally, then the resulting surface appearance is re-used when calculating the reflection.\n\nIf you wanted to avoid this, you would need to adjust the shading of the ground to exclude the spheres - but only when it is being used in a reflection.\n\nThis can be done using the aiRaySwitchShader node. Coupled with an aiTraceSet node on the specular reflection branch, the shading of the plane ignores the spheres so they don't cast a shadow.\n\nEnable the ShaderAssignment node whilst an IPR render is running to see the difference." )
```

Comment:
### Backdrop text
When ~~inclusive~~ **the Inclusive plug** is turned off on the ~~aiTraceSet~~ **TraceSet** node, the spheres wil be excluded from the reflections, but their shadows still appear on the ground.

This is because the plane is shaded normally, ~~then the resulting~~ **and its rendered** surface appearance is re-used when calculating the reflection.

If you wanted to avoid this, you would need to adjust the shading of the ~~ground~~ **plane** to exclude the spheres - but only when it is being used in ~~a~~ **the** reflection.

~~This can be done~~ **We work around this by** using the ~~aiRaySwitchShader~~ **RaySwitchShader** node **(Arnold > Shader > Utility > Ray Switch Shader)**. Coupled with ~~an aiTraceSet~~ **another TraceSet** node on the specular reflection branch **of this shader sub-network**, the shading of the plane ignores the spheres so they don't cast a shadow.

~~Enable~~ **Try enabling** the ShaderAssignment node whilst an IPR render is running to see the difference.

### Comment
Probably a good idea to rename the ShaderAssignment node to ShaderAssignment_Plane.

It's also probably worth mentioning that this shader network also completely replaces the plane's initial shader.

Reply:
Good idea. I've updated the text, but retained a few of the original terms for technical or contextual accuracy: 

> When inclusive plug is turned off on the TraceSet node, the spheres wil be excluded from the reflections, but their shadows still appear on the ground.
>
> This is because the plane is shaded normally, and the resulting surface appearance is re-used when calculating the reflection.
>
> If you wanted to avoid this, you would need to adjust the shading of the plane to exclude the spheres - but only when it is being used in a reflection.
>
> This can be achieved by using a RaySwitchShader node (Arnold > Shader > Utility > Ray Switch Shader) in the plane's shading network. Coupled with a TraceSet node (configured to exclude the spheres) in the specular reflection branch, the shading of the plane then ignores the spheres, so they don't cast a shadow.
>
>Try enabling the ShaderAssignment_Plane node whilst an IPR is running to see the difference.